### PR TITLE
Add info & code for bootstrapping realm with initial subscription

### PR DESCRIPTION
**Pull request by MongoCaleb**
### Jira

https://jira.mongodb.org/browse/DOCSP-22536

### Staged Changes (Requires MongoDB Corp SSO)

https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/bootstrap/sdk/dotnet/examples/flexible-sync/#add-a-query-to-the-set-of-subscriptions

### CODE REVIEW

File: source/sdk/dotnet/examples/flexible-sync.txt [hunk 1]
```diff
@@ -98,6 +98,27 @@ update existing subscriptions, as shown in the examples below.
...
+
+Bootstrap the Realm with Initial Subscriptions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can bootstrap a realm with an initial subscription set
+when you open it with a :dotnet-sdk:`FlexibleSyncConfiguration  
+<reference/Realms.Sync.FlexibleSyncConfiguration.html>`. Pass the 
+:dotnet-sdk:`PopulateInitialSubscriptions 
+<reference/Realms.Sync.FlexibleSyncConfiguration.html#Realms_Sync_FlexibleSyncConfiguration_PopulateInitialSubscriptions>` 
+parameter with the subscription queries you want to use to bootstrap the realm:
```
Comment (nirinchev) [inline source/sdk/dotnet/examples/flexible-sync.txt:113]: This sentence is not applicable to the API exposed by the .NET SDK. A more precise wording is something like: `Set the ...PopulateInitialSubscriptions... property to a callback that will be invoked when the Realm is first created to add the queries you want to use to bootstrap the realm`.

File: source/sdk/dotnet/examples/flexible-sync.txt [hunk 2]
```diff
@@ -98,6 +98,27 @@ update existing subscriptions, as shown in the examples below.
 
 Add a Query to the Set Of Subscriptions
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You must have at least one subscription before you can read from or write
+to the realm. You can create an initial subscription when you configure Flexible 
+Sync, or you can add subscriptions after initialization.
```
Comment (nirinchev) [inline source/sdk/dotnet/examples/flexible-sync.txt:103]: [nit] "You can create an initial subscription" implies that it has to be one, when in reality there's no such restriction/recommendation.

**Review by dacharyc — APPROVED** [review]
Tech review comments notwithstanding, the copy LGTM and example makes sense. (In fact, I love the example.)

File: source/examples/generated/dotnet/Guitar.codeblock.guitar-object.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    public string Make { get; set; }
+
+    public string Model { get; set; }
+
+    public double Price { get; set; }
+
+    public string Owner { get; set; }
+
+    public Guitar()
```
Comment (papafe) [inline source/examples/generated/dotnet/Guitar.codeblock.guitar-object.cs:14]: You don't need to have an empty constructor

File: source/examples/generated/dotnet/Program.codeblock.read-filter-sort.cs [hunk 4]
```diff
@@ -0,0 +1,2 @@
+var allGuitars = realm.All<Guitar>();
+  var lessExpensiveGuitars = realm.All<Guitar>().Where(g => g.Price < 400);    var guitarsSortedByMake = realm.All<Guitar>().OrderBy(g => g.Make      var specifiGuitarById = realm.Find<Guitar>(someGuitarId);
```
Comment (papafe) [inline source/examples/generated/dotnet/Program.codeblock.read-filter-sort.cs:2]: I'm not sure where is this used, but `guitarSortedByMake` seems incomplete

**Review by papafe — COMMENTED** [review]
Overall the text makes sense for `PopulateInitialSubscriptions` makes sense.
I've left some notes for the code regarding guitars, even though I'm not sure where is that used. 
Also, there are several code style warnings that could be addressed (nothing major though)

**Pull request merged**